controllers/activemqartemis_controller_test.go

```diff
@@ -122,6 +122,236 @@ var _ = Describe("artemis controller", func() {
+		It("default user credential secret", func() {
+			By("deploy a broker")
+			brokerCr, createdBrokerCr := DeployCustomBroker(defaultNamespace, nil)
+
+			By("checking the default credential secret created")
+			credSecretKey := types.NamespacedName{
+				Name:      brokerCr.Name + "-credentials-secret",
+				Namespace: defaultNamespace,
+			}
+			credSecret := &corev1.Secret{}
+			createdSs := &appsv1.StatefulSet{}
+			ssKey := types.NamespacedName{Name: namer.CrToSS(brokerCr.Name), Namespace: defaultNamespace}
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, credSecretKey, credSecret)).Should(Succeed())
+				g.Expect(len(credSecret.Data)).To(Equal(4))
+				g.Expect(len(credSecret.OwnerReferences) > 0).Should(BeTrue())
+				ownerFound := false
+				for _, oref := range credSecret.OwnerReferences {
+					if oref.Kind == "ActiveMQArtemis" && oref.Name == brokerCr.Name {
+						ownerFound = true
+						break
+					}
+				}
+				g.Expect(ownerFound).To(BeTrue())
+
+				g.Expect(k8sClient.Get(ctx, ssKey, createdSs)).Should(Succeed())
+
+				initContainer := createdSs.Spec.Template.Spec.InitContainers[0]
+				userFound := false
+				passwordFound := false
+				clusterUserFound := false
+				clusterPasswordFound := false
+				for _, envVar := range initContainer.Env {
+					if envVar.Name == "AMQ_USER" {
+						userFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_USER"), envVar)
+					}
+					if envVar.Name == "AMQ_PASSWORD" {
+						passwordFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_PASSWORD"), envVar)
+					}
+					if envVar.Name == "AMQ_CLUSTER_USER" {
+						clusterUserFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_CLUSTER_USER"), envVar)
+					}
+					if envVar.Name == "AMQ_CLUSTER_PASSWORD" {
+						clusterPasswordFound = true
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Name).To(Equal(brokerCr.Name+"-credentials-secret"), envVar)
+						g.Expect(envVar.ValueFrom.SecretKeyRef.Key).To(Equal("AMQ_CLUSTER_PASSWORD"), envVar)
+					}
+				}
+				g.Expect(userFound).To(BeTrue())
+				g.Expect(passwordFound).To(BeTrue())
+				g.Expect(clusterUserFound).To(BeTrue())
+				g.Expect(clusterPasswordFound).To(BeTrue())
+
+			}, timeout, interval).Should(Succeed())
+
+			// cleanup
+			Expect(k8sClient.Delete(ctx, brokerCr)).Should(Succeed())
+			By("check it has gone")
+			Eventually(func() bool {
+				return checkCrdDeleted(brokerCr.Name, defaultNamespace, createdBrokerCr)
+			}, timeout, interval).Should(BeTrue())
+		})
+
+		It("user credential secret", func() {
```

Comment:
my suspicion is that in this test there are errors in the logs, b/c we track this secret and try and create it, but it will already exist.

Reply:
it is true.